### PR TITLE
config sync changes and rename

### LINKED ISSUE #1716: [v2] Multiple crashes of the router pod before network stabilizes
**Describe the bug**
Using a scripted two-site demonstration setup (script included below), the sites initialize but the router pods go into crash-loop-backoff and restart twice before the network finally stablizes.

**How To Reproduce**
On a cluster, create namespaces demo-dmz-a and demo-dmz-b and run the provided setup script.  Watch the pods and sites to observe crashes prior to network stabilization.

**Expected behavior**
I expect the network to stabilize in an orderly fashion without seeing crash indications.

**Environment details**
 - Skupper CLI: None used
 - Skupper Operator (if applicable): head of the v2 branch
 - Platform: openshift

**Additional context**
The error seen in the router log prior to crashing:
```
2024-10-11 16:02:24.121816 +0000 ROUTER (critical) Router start-up failed: Python: CError: Configuration: Failed to configure TLS caCertFile '/etc/skupper-router-certs/skupper-site-server/ca.crt' from sslProfile 'skupper-site-server'
```

The script used to reproduce:
```bash
NS_PREFIX=demo

for i in dmz-a dmz-b; do
echo Create site $i in namespace ${NS_PREFIX}-$i
cat >temp <<EOF
apiVersion: skupper.io/v1alpha1
kind: Site
metadata:
  name: $i
spec:
  routerMode: "interior"
  ha: false
---
apiVersion: skupper.io/v1alpha1
kind: RouterAccess
metadata:
  name: $i-peer
spec:
  generateTlsCredentials: true
  issuer: skupper-site-ca
  roles:
  - name: inter-router
    port: 55671
  tlsCredentials: skupper-site-server
---
apiVersion: skupper.io/v1alpha1
kind: AccessGrant
metadata:
  name: $i-grant
spec:
  redemptionsAllowed: 10
  expirationWindow: 1h
EOF
kubectl apply -n ${NS_PREFIX}-$i -f temp

echo Generating access token token-$i.yaml
kubectl wait --for=condition=ready accessgrant/$i-grant -n ${NS_PREFIX}-$i
kubectl get accessgrant $i-grant -n ${NS_PREFIX}-$i -o yaml > temp
URL=`cat temp | yq '.status.url'`
CODE=`cat temp | yq '.status.code'`
CA=`cat temp | yq -r '.status.ca' | awk '{ print "    " $0 }'`
cat >token-$i.yaml <<EOF
apiVersion: skupper.io/v1alpha1
kind: AccessToken
metadata:
  name: $i-token
spec:
  url: ${URL}
  code: ${CODE}
  ca: |
${CA}
EOF

rm temp
echo
done

echo Link dmz-b to dmz-a
kubectl apply -f token-dmz-a.yaml -n ${NS_PREFIX}-dmz-b
```

### PR DESCRIPTION
**Pull request by grs**
* add init mode and use that in init-container to fix #1716 
* rename image to kube-adaptor to reflect its wider role

### CODE REVIEW

**Comment by ssorj** [comment]
I like the name kube-adaptor, but so far I think "router-adaptor" is more helpful in the context of its sibling binaries and images.  I came to this conclusion by thinking about how the names would look in the source tree and the quay org.

Screenshot from 2024-10-30 06-39-22

Screenshot from 2024-10-30 06-42-26

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 VERSION := $(shell git describe --tags --dirty=-modified --always)
 CONTROLLER_IMAGE := quay.io/skupper/controller:v2-latest
 BOOTSTRAP_IMAGE := quay.io/skupper/bootstrap:v2-latest
-CONFIG_SYNC_IMAGE := quay.io/skupper/config-sync:v2-latest
+CONFIG_SYNC_IMAGE := quay.io/skupper/kube-adaptor:v2-latest
```
Comment (fgiorgetti) [inline Makefile:4]: I believe that the variable must also be renamed to: `ADAPTOR_IMAGE` (based on circleci changes).

Comment (grs) [reply Makefile]: Thanks! Well spotted! I have pushed a fix for that now.

File: internal/kube/adaptor/config_init.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	paths "path"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/skupperproject/skupper/pkg/qdr"
+)
+
+func InitialiseConfig(client kubernetes.Interface, namespace string, path string, routerConfigMap string) error {
+	ctxt := context.Background()
```
Comment (fgiorgetti) [inline internal/kube/adaptor/config_init.go:15]: The config-init container logs are basically empty.
Maybe adding some basic info into it, just to see sslProfile secrets being syncronized, would be nice.

**Review by fgiorgetti — APPROVED** [review]

**Review by c-kruse — APPROVED** [review]

**Comment by nluaces** [comment]
Helm chart needs to be updated as well:
https://github.com/skupperproject/skupper/blob/81a9f91f61a8c95510e14abe869c82aa29aa25e1/deployments/helm/values.yaml#L3
https://github.com/skupperproject/skupper/blob/81a9f91f61a8c95510e14abe869c82aa29aa25e1/deployments/helm/templates/deploy_controller.yaml#L241

**Comment by grs** [comment]
I've separated the fix for #1716 from any naming change here  (#1769) to get that in while the final name is debated. Hence closing this.

**Pull request closed without merging**